gaitmesh_ros/include/gaitmesh_ros/common.hpp

```diff
@@ -26,5 +32,17 @@ inline bool pointsOnSameSideOfLine(const Eigen::Vector3d& p1, const Eigen::Vecto
 
 inline bool pointInTriangle(const Eigen::Vector3d& p, const Eigen::Vector3d& a, const Eigen::Vector3d& b, const Eigen::Vector3d& c)
 {
+    if ((p - a).norm() < 1e-3 || (p - b).norm() < 1e-3 || (p - c).norm() < 1e-3)
```

Comment:
We should add a comment here (since the 1e-3 is a selected threshold) that if the query point coincides with one of the vertices within a set threshold (here 1mm), we consider it to be within the triangle.